signer/src/block_observer.rs

```diff
@@ -570,13 +570,21 @@ impl<C: Context, B> BlockObserver<C, B> {
         };
 
         let rolling_limits = limits.rolling_withdrawal_limits();
+        let context_window = rolling_limits.blocks.min(u16::MAX as u64) as u16;
```

Comment:
Do we have any concerns about performance if/when the context window is `u16::MAX`?

Reply:
Hmmm, this should only happen if the API returns a blocks value that is greater than `u16::MAX`. But this is a footgun for sure. Maybe we cap it at 2 weeks worth of blocks or something?

Reply:
We could probably use the context window from the config as a backup? I mean, that's already a handicap for a lot of the other operations.

Reply:
Having this as a variable does seems useful to me, eg if we need to increase the rolling cap to let a certain amount go through we can still tweak the block numbers instead of having to raise and then lower the cap (since the time scale is fixed).
I'd vote to keep `u16::MAX` right now: the performance implications are negligible imho, in the sense that unless we mess up setting it on emily it shouldn't happen, and even if it does it's once per bitcoin block so not that bad regardless. I would be fine with using an even "safer" lower amount but it's probably not worth it spending the time to come up with a new number; maybe 50k? (a year-ish).

Reply:
I'm fine either way since yeah, this _shouldn't_ be a relevant scenario, I just wanted to raise the Q as ~65k felt very unnecessary. Hopefully we've implemented some sort of pruning before we've reached any sort of block levels where performance would be a real issue, though.

Reply:
Let's go with changing the type.

Reply:
Updated in https://github.com/stacks-network/sbtc/pull/1526/commits/91e3561b783402969fc034c6133a7b54250bfcd7

Reply:
Yeah that feels a little better tbh 👍

Reply:
Yeah I agree. I had it in some commit at one point and had removed it before pushing.